rct_optimizations/src/rct_optimizations/dh_chain_kinematic_calibration.cpp

```diff
@@ -219,6 +220,222 @@ KinematicCalibrationResult optimize(const KinematicCalibrationProblem2D3D &param
   result.converged = summary.termination_type == ceres::CONVERGENCE;
   result.final_cost_per_obs = summary.final_cost / summary.num_residuals;
 
+
+  // Save the transforms
+  result.camera_mount_to_camera = createTransform(t_cm_to_c, aa_cm_to_c);
+  result.target_mount_to_target = createTransform(t_tm_to_t, aa_tm_to_t);
+  result.camera_base_to_target_base = createTransform(t_ccb_to_tcb, aa_ccb_to_tcb);
+
+  // Save the DH parameter offsets
+  if (params.camera_chain.dof() == 0)
+    result.camera_chain_dh_offsets = Eigen::MatrixX4d::Zero(params.camera_chain.dof(), 4);
+  else
+    result.camera_chain_dh_offsets = camera_chain_dh_offsets;
+
+  if (params.target_chain.dof() == 0)
+    result.target_chain_dh_offsets = Eigen::MatrixX4d::Zero(params.target_chain.dof(), 4);
+  else
+    result.target_chain_dh_offsets = target_chain_dh_offsets;
+
+  ceres::Covariance::Options cov_options = rct_optimizations::DefaultCovarianceOptions();
+  cov_options.null_space_rank = -1;  // automatically drop terms below min_reciprocal_condition_number
+  result.covariance = computeCovariance(problem, std::vector<const double*>(parameters.begin(), parameters.end()), param_labels, cov_options);
+
+  return result;
+}
+
+KinematicCalibrationResult optimize(const KinematicCalibrationProblemPose3D &params)
+{
+  // Initialize the optimization variables
+  // Camera mount to camera (cm_to_c) quaternion and translation
+  Eigen::Vector3d t_cm_to_c(params.camera_mount_to_camera_guess.translation());
+  Eigen::AngleAxisd rot_cm_to_c(params.camera_mount_to_camera_guess.rotation());
+  Eigen::Vector3d aa_cm_to_c(rot_cm_to_c.angle() * rot_cm_to_c.axis());
+
+  std::array<std::string, 3> t_cm_to_c_labels({params.label_camera_mount_to_camera + "_x",
+                                               params.label_camera_mount_to_camera + "_y",
+                                               params.label_camera_mount_to_camera + "_z"});
+  std::array<std::string, 3> aa_cm_to_c_labels({params.label_camera_mount_to_camera + "_rx",
+                                                params.label_camera_mount_to_camera + "_ry",
+                                                params.label_camera_mount_to_camera + "_rz"});
+
+  // Target mount to target (cm_to_c) quaternion and translation
```

Comment:
I think this comment in both `optimize` functions is wrong; it should say `(tm_to_t) translation and angle-axis`